packages/nextcloud/test/litmus_test.dart

```diff
@@ -0,0 +1,246 @@
+/// These tests try to replicate all tests in https://github.com/notroj/litmus that are applicable to Nextcloud (e.g. locks are not supported).
+/// Some tests are not applicable because the client already solves some of the problems.
+@Retry(3)
+library litmus_test;
+
+import 'dart:convert';
+import 'dart:math';
+import 'dart:typed_data';
+
+import 'package:nextcloud/nextcloud.dart';
+import 'package:test/test.dart';
+
+import 'helper.dart';
+
+void main() {
+  group('litmus', () {
+    late DockerImage image;
+    setUpAll(() async => image = await getDockerImage());
+
+    late DockerContainer container;
+    late TestNextcloudClient client;
+    setUp(() async {
+      container = await getDockerContainer(image);
+      client = await getTestClient(container);
+    });
+    tearDown(() => container.destroy());
+
+    group('basic', () {
+      test('options', () async {
+        final options = await client.webdav.options();
+        expect(options.capabilities, contains('1'));
+        expect(options.capabilities, contains('3'));
+        // Nextcloud only contains a fake plugin for Class 2 support: https://github.com/nextcloud/server/blob/master/apps/dav/lib/Connector/Sabre/FakeLockerPlugin.php
+        // It does not actually support locking and is only there for compatibility reasons.
+        expect(options.capabilities, isNot(contains('2')));
+      });
+
+      for (final (name, path) in [
+        ('put_get', 'res'),
+        ('put_get_utf8_segment', 'res-%e2%82%ac'),
+      ]) {
+        test(name, () async {
+          final content = Uint8List.fromList(utf8.encode('This is a test file'));
+
+          final response = await client.webdav.put(content, path);
+          expect(response.statusCode, 201);
+
+          final downloadedContent = await client.webdav.get(path);
+          expect(downloadedContent, equals(content));
+        });
+      }
+
+      test('put_no_parent', () async {
+        expect(
+          () => client.webdav.put(Uint8List(0), '409me/noparent.txt'),
+          // https://github.com/nextcloud/server/issues/39625
+          throwsA(predicate((final e) => (e! as DynamiteApiException).statusCode == 409)),
```

Comment:
```suggestion
          throwsA(predicate<DynamiteApiException>((final e) => e.statusCode == 409)),
```

Same for all the other tests